math/mathcore/inc/TRandom3.h

```diff
@@ -24,6 +24,19 @@
 
 #include "TRandom.h"
 
+#include <array>
+
+class TRandom3;
+
+class TRandom3State {
+   friend TRandom3;
+   std::array<UInt_t,624> fMt;
+   Int_t fCount624;
+   TRandom3State(const UInt_t *mt, Int_t count624) : fCount624(count624) {
+      std::copy_n(mt, 624, fMt.begin());
+   }
+};
```

Comment:
How about putting this class inside TRandom3 itself? That way the details of the state stay opaque and can be changed later if needed.

Reply:
what do you think @lmoneta ?